synapseclient/core/exceptions.py

```diff
@@ -57,13 +65,35 @@ class SynapseUploadAbortedException(SynapseError):
     """Raised when a worker thread detects the upload was
     aborted and stops further processing."""
 
-    pass
-
 
 class SynapseUploadFailedException(SynapseError):
     """Raised when an upload failed. Should be chained to a cause Exception"""
 
-    pass
+
+def _get_message(response: httpx.Response) -> Union[str, None]:
+    """Extracts the message body or a response object by checking for a json response
+    and returning the reason otherwise getting body.
+    """
+    try:
+        if utils.is_json(response.headers.get("content-type", None)):
+            json = response.json()
+            return json.get("reason", None)
+        else:
+            # if the response is not JSON, return the text content
+            return response.text
+    except (AttributeError, ValueError):
+        # The response can be truncated. In which case, the message cannot be retrieved.
+        return None
```

Comment:
I'm not sure how I feel about silently returning `None` for calls to the API. Can we add a warning message here saying that an error was received, likely due to a truncated response?